ddemangle.d

```diff
@@ -58,10 +60,32 @@ const(char)[] demangleMatch(T)(Captures!(T) m)
     }
 }
 
-auto ddemangle(T)(T line)
-    if (is(T : const(char)[]))
+const(char)[] demangleMatchUnderscoreMissing(T)(Captures!(T) m)
+        if (is(T : const(char)[]))
 {
-    return replaceAll!(demangleMatch)(line, reDemangle);
+    static Appender!string ret;
+    ret.ret;
+    ret ~= "_";
+    ret ~= m.hit;
+    auto line2 = ret.data;
+    auto result = demangle(line2);
+    if (result == line2)
+    {
+        // Demangling failed, return original match
+        return m.hit;
+    }
+    else
+    {
+        return result;
+    }
+}
+
+auto ddemangle(T)(T line, bool underscore_missing) if (is(T : const(char)[]))
```

Comment:
style:

```
auto ddemangle(T)(T line, bool underscore_missing)
if (is(T : const(char)[]))
```

also no snake_case and `bool` flags, use `std.typecons.Flag`

Reply:
done